### ISSUE TITLE
White materials

### ISSUE THREAD

**Issue by Anony3lk** [FlailingFog/KK-Blender-Porter-Pack#300]
Everytime I try to import a KK model that uses the Face 3 or Face 2 type the model doesn't have its rigging and its materials are all white.

Example:
https://user-images.githubusercontent.com/116595159/197626392-2e6dec54-f0d0-49a8-8f7a-b5b5ee83937b.PNG
https://user-images.githubusercontent.com/116595159/197626403-47988449-6aca-426b-8968-7fcae6f4dba3.PNG

Import settings:
![Import settings](https://user-images.githubusercontent.com/116595159/197626452-c9eb37ad-0727-4b01-b1d4-f7a0683b132d.PNG)

I'm using KKBP v6.1.1, Cats v.0.19.0, KKBP_Exporter v3.94, and Blender v3.3.1.

Here's the card:
![Koikatu_F_20221024005100851_Liza Stella](https://user-images.githubusercontent.com/116595159/197626633-9ba6f9c7-b0d0-4d2f-a5bb-c7d2db879bbc.png)

Here's the whole console log for when importing

> Read prefs: C:\Users\Administrator\AppData\Roaming\Blender Foundation\Blender\3.3\config\userpref.blend
Reloading external rigs...
Reloading external metarigs...
>### Loading CATS...
>mmd_tools_local registed 131 classes
>### Loaded CATS successfully!
>ValueError: the return value must be None
File "C:\Users\Administrator\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\cats-blender-plugin-master\tools\settings.py", line 249, in update_settings
====    KKBP Log    ====
Info: Imported MMD model from "C:\Users\Administrator\Documents\Blender\Modeling\Models\HF Liza 3.94 (V,H,S)\model.pmx"
Info: Imported MMD model from "C:\Users\Administrator\Documents\Blender\Modeling\Models\HF Liza 3.94 (V,H,S)\Outfit 05\model.pmx"
Finalizing PMX file...
Modifying armature...
Finished in 2.17s
Translating and combining shapekeys...
Warning: 1 x Draw window and swap: 35.1339 ms, average: 35.13390000 ms
Warning: 1 x Draw window and swap: 7.2068 ms, average: 7.20679993 ms
Warning: 1 x Draw window and swap: 11.6657 ms, average: 11.66570000 ms
Warning: 1 x Draw window and swap: 15.0887 ms, average: 15.08869999 ms
Warning: 1 x Draw window and swap: 18.3050 ms, average: 18.30500003 ms
Warning: 1 x Draw window and swap: 13.7527 ms, average: 13.75269995 ms
Warning: 1 x Draw window and swap: 16.0906 ms, average: 16.09059994 ms
Warning: 1 x Draw window and swap: 16.0200 ms, average: 16.01999998 ms
Warning: 1 x Draw window and swap: 15.6678 ms, average: 15.66779998 ms
Warning: 1 x Draw window and swap: 15.9285 ms, average: 15.92850010 ms
Warning: 1 x Draw window and swap: 16.0045 ms, average: 16.00450010 ms
Warning: 1 x Draw window and swap: 22.7165 ms, average: 22.71649998 ms
Warning: 1 x Draw window and swap: 9.8666 ms, average: 9.86659992 ms
Warning: 1 x Draw window and swap: 15.7046 ms, average: 15.70460002 ms
Warning: 1 x Draw window and swap: 16.0957 ms, average: 16.09569998 ms
Warning: 1 x Draw window and swap: 15.9037 ms, average: 15.90370003 ms
Warning: 1 x Draw window and swap: 16.0913 ms, average: 16.09129994 ms
Warning: 1 x Draw window and swap: 16.2175 ms, average: 16.21749997 ms
Warning: 1 x Draw window and swap: 15.8849 ms, average: 15.88490000 ms
Warning: 1 x Draw window and swap: 16.0499 ms, average: 16.04989998 ms
Warning: 1 x Draw window and swap: 16.2190 ms, average: 16.21900010 ms
Warning: 1 x Draw window and swap: 15.9031 ms, average: 15.90310002 ms
Warning: 1 x Draw window and swap: 16.0199 ms, average: 16.01989998 ms
Warning: 1 x Draw window and swap: 16.1995 ms, average: 16.19950007 ms
Warning: 1 x Draw window and swap: 15.9386 ms, average: 15.93860006 ms
Warning: 1 x Draw window and swap: 17.8403 ms, average: 17.84029999 ms
Warning: 1 x Draw window and swap: 14.3903 ms, average: 14.39030003 ms
Warning: 1 x Draw window and swap: 16.2186 ms, average: 16.21859998 ms
Warning: 1 x Draw window and swap: 15.4268 ms, average: 15.42680000 ms
Warning: 1 x Draw window and swap: 16.4621 ms, average: 16.46210009 ms
Warning: 1 x Draw window and swap: 15.6922 ms, average: 15.69219993 ms
Warning: 1 x Draw window and swap: 16.5976 ms, average: 16.59760007 ms
Warning: 1 x Draw window and swap: 16.1695 ms, average: 16.16949996 ms
Warning: 1 x Draw window and swap: 106.9091 ms, average: 106.90909997 ms
Warning: 1 x Draw window and swap: 87.9725 ms, average: 87.97249990 ms
Warning: 1 x Draw window and swap: 46.1313 ms, average: 46.13130004 ms
Finished in 1.29s
Separating body, clothes, hair, hitboxes and shadowcast, then removing duplicate materials...
Warning:        Somehow found a false duplicate material but didn't merge: cf_m_tang.001
Moving duplicate material Bonelyfans in slot 20 to the original slot 0
Moving duplicate material cf_m_namida_00 in slot 8 to the original slot 7
Moving duplicate material cf_m_namida_00 in slot 9 to the original slot 7
Moving duplicate material cf_m_sirome_00 in slot 11 to the original slot 10
Info: Removed 4 slots
Warning:        Somehow found a false duplicate material but didn't merge: o_hit_armL_M.001
Warning:        Somehow found a false duplicate material but didn't merge: o_hit_armR_M.001
Warning:        Somehow found a false duplicate material but didn't merge: o_hit_footL_M.001
Warning:        Somehow found a false duplicate material but didn't merge: o_hit_footR_M.001
Warning:        Somehow found a false duplicate material but didn't merge: o_hit_handL_M.001
Warning:        Somehow found a false duplicate material but didn't merge: o_hit_handR_M.001
Warning:        Somehow found a false duplicate material but didn't merge: o_hit_hara_M.001
Warning:        Somehow found a false duplicate material but didn't merge: o_hit_haraB_M.001
Warning:        Somehow found a false duplicate material but didn't merge: o_hit_haraUnder_M.001
Warning:        Somehow found a false duplicate material but didn't merge: o_hit_kneeBL_M.001
Warning:        Somehow found a false duplicate material but didn't merge: o_hit_kneeBR_M.001
Warning:        Somehow found a false duplicate material but didn't merge: o_hit_kneeL_M.001
Warning:        Somehow found a false duplicate material but didn't merge: o_hit_kneeR_M.001
Warning:        Somehow found a false duplicate material but didn't merge: o_hit_kokan_M.001
Warning:        Somehow found a false duplicate material but didn't merge: o_hit_legBL_M.001
Warning:        Somehow found a false duplicate material but didn't merge: o_hit_legBR_M.001
Warning:        Somehow found a false duplicate material but didn't merge: o_hit_legL_M.001
Warning:        Somehow found a false duplicate material but didn't merge: o_hit_legR_M.001
Warning:        Somehow found a false duplicate material but didn't merge: o_hit_mune_M.001
Warning:        Somehow found a false duplicate material but didn't merge: o_hit_muneB_M.001
Warning:        Somehow found a false duplicate material but didn't merge: o_hit_siriL_M.001
Warning:        Somehow found a false duplicate material but didn't merge: o_hit_siriR_M.001
Warning:        Somehow found a false duplicate material but didn't merge: cf_O_face_atari_M.001
Moving duplicate material cf_m_hair_b_114_00 3965 in slot 1 to the original slot 0
Moving duplicate material cf_m_hair_b_114_00 3965 in slot 2 to the original slot 0
Moving duplicate material cf_m_hair_b_114_00 3965 in slot 3 to the original slot 0
Moving duplicate material cf_m_hair_b_114_00 3965 in slot 4 to the original slot 0
Moving duplicate material cf_m_hair_f_1_00 4285 in slot 6 to the original slot 5
Moving duplicate material cf_m_hair_f_1_00 4285 in slot 7 to the original slot 5
Moving duplicate material cf_m_hair_f_1_00 4285 in slot 8 to the original slot 5
Moving duplicate material cf_m_hair_f_1_00 4285 in slot 9 to the original slot 5
Moving duplicate material Shirt_done 8935 in slot 36 to the original slot 35
Moving duplicate material Shirt_done 8965 in slot 38 to the original slot 37
Moving duplicate material Shirt_done 8965 in slot 39 to the original slot 37
Moving duplicate material olga_pants 9035 in slot 41 to the original slot 40
Moving duplicate material olga_pants 9065 in slot 43 to the original slot 42
Info: Removed 13 slots
Error:          Unknown python error occurred
Traceback (most recent call last):
  File "C:\Users\Administrator\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\KK-Blender-Porter-Pack-V611\importing\separatebody.py", line 615, in execute
    separate_everything(context)
  File "C:\Users\Administrator\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\KK-Blender-Porter-Pack-V611\importing\separatebody.py", line 232, in separate_everything
    if bpy.data.objects['Outfit 00'].material_slots.get('cf_O_face_atari_M.001'):
KeyError: 'bpy_prop_collection[key]: key "Outfit 00" not found'
Error: Traceback (most recent call last):
  File "C:\Users\Administrator\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\KK-Blender-Porter-Pack-V611\importing\separatebody.py", line 615, in execute
    separate_everything(context)
  File "C:\Users\Administrator\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\KK-Blender-Porter-Pack-V611\importing\separatebody.py", line 232, in separate_everything
    if bpy.data.objects['Outfit 00'].material_slots.get('cf_O_face_atari_M.001'):
KeyError: 'bpy_prop_collection[key]: key "Outfit 00" not found'
Traceback (most recent call last):
  File "C:\Users\Administrator\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\KK-Blender-Porter-Pack-V611\importing\importbuttons.py", line 89, in execute
    bpy.ops.kkb.separatebody('INVOKE_DEFAULT'),
  File "C:\Program Files\Blender Foundation\Blender 3.3\3.3\scripts\modules\bpy\ops.py", line 111, in __call__
    ret = _op_call(self.idname_py(), C_dict, kw, C_exec, C_undo)
RuntimeError: Error: Traceback (most recent call last):
  File "C:\Users\Administrator\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\KK-Blender-Porter-Pack-V611\importing\separatebody.py", line 615, in execute
    separate_everything(context)
  File "C:\Users\Administrator\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\KK-Blender-Porter-Pack-V611\importing\separatebody.py", line 232, in separate_everything
    if bpy.data.objects['Outfit 00'].material_slots.get('cf_O_face_atari_M.001'):
KeyError: 'bpy_prop_collection[key]: key "Outfit 00" not found'

**Comment by BTHP5** [comment]
Are you using the KKBP exporter plugin 3.5? That's the version that works with Koikatsu.
Also, I recommend using Blender version 3.3.1
Also, sometimes, the exporter has exported two .pmx, one in the main character folder, and one in the "Outfit 00" folder, do not export the one in the outfit folder.

**Comment by BTHP5** [comment]
Also, the plugin has since been updated, so I recommend using KKBP 6.6.1, and the newest KKBP exporter along with it :]

**Comment by Anony3lk** [comment]
Thanks for the suggestions. 

I'm actually using KKBP 6.6.1, my dumbass just wrote 6.6.0 (I'll edit it to fix that), and I'm also using the latest exporter (v.3.94). I reinstalled both just to make sure as well.

Also, I haven't been using the model.pmx files found in the Outfit folders for importing into blender. I've also tried to import them to make sure a file wasn't saved in a wrong location or something but to no avail.

To clarify, I can successfullly import models with the Face 1 types from KK. So, unless you've got any more ideas, I think something might just be wrong with my KK :p

And I honestly don't want to try to reinstall KK right now, so I'll make do with what I can export and import and reinstall KK in the future when I really want to try that option. I'll reply to this post (or repost) if reinstalling KK fixes this problem, since others might be experiencing the same thing. I'll also try anything else anyone suggests and report back.
